### PR TITLE
fix: Do not take a snapshot before the interval has been set from the…

### LINKED ISSUE #6266: Snapshot: can't load from snapshot taken at height 1 after checkpoint reload
# Problem encountered
Which is ridiculous because there is no value in restoring from a snapshot at height 1 anyway. The reason it fails is because of the epoch-fast forwarding we do after restoring from a checkpoint which happens at the start of block 2 and we don't save in the snapshot that we need to fast forward.

I think what I'd rather do than handle this case, is to just not take a snapshot at height 1 because I have always believed this is a bug anyway. If you set the snapshot interval to 1000 you'd expect a snapshot at height 1000, 2000, etc and not 1001, 2001. 

Its just a consequence of the snapshot engine setting its `interval` field before the network parameters are propagated.

# Observed behaviour
A clear and concise description of how the system is behaving.

# Expected behaviour

# System response
Describe what the system response was, include the output from the command, automation, or else.

# Steps to reproduce

## Manual
Steps to reproduce the behaviour manually:

## Automation
Link to automation and explanation on how to run it to reproduce the problem/bug

# Evidence

## Logs
If applicable, add logs and/or screenshots to help explain your problem.

## Additional context
Add any other context about the problem here including; system version numbers, components affected.

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by ze97286**
… network parameter

Closes #6266

### CODE REVIEW

**Review by jeremyletang — DISMISSED** [review]

**Comment by wwestgarth** [comment]
This is no way fixes the problem. If you look in the logs of the system test, with an interval of 12 we are still taking a snapshots at height 1, 13, 25 and not 12, 24.

**Comment by ze97286** [comment]
> This is no way fixes the problem. If you look in the logs of the system test, with an interval of 12 we are still taking a snapshots at height 1, 13, 25 and not 12, 24.

you say in the ticket that the issue is that we're taking snapshot before the interval is propagated from the net param, this PR fixes that. why would it take a snapshot at block 1?

**Review by ValentinTrinque — DISMISSED** [review]

**Comment by wwestgarth** [comment]
I said in the ticket:
```
Its just a consequence of the snapshot engine setting its interval field before the network parameters are propagated.
```
and due to `interval` being set to `1` at initialisation, when the netparameter comes in we do not increase it due to the logic you just found.

Regardless, the ticket was to stop taking a snapshot at height 1, which you didn't fix because you didn't run a network to verify, or write a unittest.

File: core/snapshot/engine.go [hunk 1]
```diff
@@ -1111,7 +1111,7 @@ func (e *Engine) Close() error {
 
 func (e *Engine) OnSnapshotIntervalUpdate(ctx context.Context, interval int64) error {
```
Comment (wwestgarth) [inline core/snapshot/engine.go:1112]: Can you write a unittest for this change please?

Comment (ze97286) [reply core/snapshot/engine.go:1112]: does the system test evidence above not prove that it works?

Comment (wwestgarth) [reply core/snapshot/engine.go:1112]: It proves it works right now. It does not prove it works tomorrow.

Comment (wwestgarth) [reply core/snapshot/engine.go:1112]: If it were a more complicated fix where a unittest was extremely complicated, then providing a screen shot is sufficient, but when a unittest would be maybe 5 lines and trivial to write, then we should be writing them.

**Review by wwestgarth — APPROVED** [review]

**Pull request merged**